app/main.py

```diff
@@ -1,3 +1,17 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable, Any
+from functools import wraps
+
+
+def cache(func: Callable) -> Callable:
+    cached = {}
+
+    @wraps(func)
+    def inner(*args) -> Any:
+        if args in cached:
+            print("Getting from cache")
+            return cached[args]
+        else:
+            cached[args] = func(*args)
+            print("Calculating new result")
+            return cached[args]
```

Comment:
You can use a single return statement out of the if/else statement instead of two inside it ☀

Reply:
☀